### ISSUE TITLE
"Xtext SDK Complete" and "Xtend IDE"

### ISSUE THREAD

**Issue by LorenzoBettini** [eclipse/xtext-eclipse#101]
TL;DR

I really think that "Xtext SDK Complete" (aka xtext.sdk.feature.group) should "include" (actually "requires" in the feature.xml) "Xtend IDE" (aka xtend.sdk.feature.group) as it always did. While, currently, after repo split, this does not hold anymore.

Long Story:

If Xtext SDK Complete did not include also Xtend IDE, then a user that installs Xtext SDK complete would create an Xtext project and could not go on, since all the .xtend files would not be compilable into Java files. So this is strong requirement for the installation in the IDE.

If one wanted to have a reduced set of things for the target platform, we could provide a reduced feature, something like "Xtext Target Platform SDK" (even though xtext.redist feature should work already). However, I've never seen any problem in adding Xtext SDK in the target platform for two reasons:

1. using Oomph, I avoid duplicate use of disk space, since the bundles in the bundle pool will be reused both for the host Eclipse and for the target platform as well
2. using Maven Tycho, even if I mention xtext.sdk in the .target file, I know that Tycho will only actually download the bundles that are required by my projects (as long as they're part of a feature mentioned in the .target file), not the whole xtext.sdk feature. (on a side note, this Tycho behavior gives problems in some situations, see e.g., https://github.com/eclipse/xtext/pull/1004, but in this case it is helpful ;)

I also noted the different behavior of Xtext SDK not including Xtend IDE anymore when I was trying to update my Xtext develoment workspace with Oomph: in spite of having the p2 repositories of both Xtext and Xtend based on the latest build in Jenkins (http://services.typefox.io/open-source/jenkins//job/xtext-eclipse/job/master/lastSuccessfulBuild/artifact/build/p2-repository/features/ and http://services.typefox.io/open-source/jenkins//job/xtext-xtend/job/master/lastSuccessfulBuild/artifact/build/p2-repository/features/, respectively), no update was found. Trying to update Xtext SDK and Xtend IDE manually from Eclipse ended up with a p2 resolution error:

```
Cannot complete the install because of a conflicting dependency.
 Software being installed: Xtend IDE 2.11.0.v20160929-1448 (org.eclipse.xtend.sdk.feature.group 2.11.0.v20160929-1448)
 Software being installed: Xtext Complete SDK 2.11.0.v20160930-0901 (org.eclipse.xtext.sdk.feature.group 2.11.0.v20160930-0901)
 Only one of the following can be installed at once: 
   Xbase Library 2.11.0.v20160929-1035 (org.eclipse.xtext.xbase.lib.feature.jar 2.11.0.v20160929-1035)
   Xbase Library 2.11.0.v20160930-0901 (org.eclipse.xtext.xbase.lib.feature.jar 2.11.0.v20160930-0901)
 Cannot satisfy dependency:
   From: Xtend IDE 2.11.0.v20160929-1448 (org.eclipse.xtend.sdk.feature.group 2.11.0.v20160929-1448)
   To: org.eclipse.xtext.xbase.lib.feature.group [2.11.0.v20160929-1035]
 Cannot satisfy dependency:
   From: Xtext Complete SDK 2.11.0.v20160930-0901 (org.eclipse.xtext.sdk.feature.group 2.11.0.v20160930-0901)
   To: org.eclipse.xtext.xbase.feature.group [2.11.0.v20160930-0901]
 Cannot satisfy dependency:
   From: Xbase 2.11.0.v20160930-0901 (org.eclipse.xtext.xbase.feature.group 2.11.0.v20160930-0901)
   To: org.eclipse.xtext.xbase.lib.feature.group [2.11.0.v20160930-0901]
 Cannot satisfy dependency:
   From: Xbase Library 2.11.0.v20160929-1035 (org.eclipse.xtext.xbase.lib.feature.group 2.11.0.v20160929-1035)
   To: org.eclipse.xtext.xbase.lib.feature.jar [2.11.0.v20160929-1035]
 Cannot satisfy dependency:
   From: Xbase Library 2.11.0.v20160930-0901 (org.eclipse.xtext.xbase.lib.feature.group 2.11.0.v20160930-0901)
   To: org.eclipse.xtext.xbase.lib.feature.jar [2.11.0.v20160930-0901]
```

Indeed, both p2 repos contain a version of xbase.lib (with different version build qualifiers), and since Xtext SDK includes xbase feature (which includes xbase.lib) and Xtend IDE includes xbase.lib, then p2 can't install them both (and here "includes" really means "includes" in the feature.xml; and we know that feature inclusion means strict version matching).

In any case, as I said at the beginning, I only see drawbacks in changing the fact that Xtext SDK does not require Xtend IDE anymore.

I really think that a single p2 repo should be built with both Xtext and Xtend and restore the original requirement.

**Comment by meysholdt** [comment]
discussed this with @spoenemann and @svenefftinge, we fully agree that the Xtext SDK should include/require the Xtend SDK

**Comment by LorenzoBettini** [comment]
great! I feel relieved :)

**Comment by meysholdt** [comment]
fixed, new repo is here:
http://services.typefox.io/open-source/jenkins/job/xtext/job/master/lastSuccessfulBuild/artifact/build/p2-repository/

the repo aggregates Xtext and Xtend
